### PR TITLE
Added Refine function for symmetric matrices

### PR DESCRIPTION
**Pull request by Saanidhyavats**
#### Other comments
With reference to issue #16234 

#### Release Notes

* assumptions
  * Added a `refine_matrixelement` function

### CODE REVIEW

**Comment by Saanidhyavats** [comment]
My previous commits also came along with the new commit. How to remove them?

File: sympy/assumptions/refine.py [hunk 1]
```diff
@@ -336,6 +337,26 @@ def refine_sign(expr, assumptions):
     return expr
 
 
+def refine_matrixelement(expr, assumptions):
+"""
```
Comment (asmeurer) [inline sympy/assumptions/refine.py]: Indent the quotes for a docstring.

File: sympy/assumptions/refine.py [hunk 2]
```diff
@@ -348,5 +369,11 @@ def refine_sign(expr, assumptions):
...
     'im': refine_im,
-    'sign': refine_sign
+    'sign': refine_sign,
+    'symmetric matrice': refine_matrixelement
 }  # type: Dict[str, Callable[[Expr, Boolean], Expr]]
+
+
+
+    
+
```
Comment (asmeurer) [inline sympy/assumptions/refine.py]: Remove these.

**Comment by asmeurer** [comment]
Why did you delete dyadic.py?

**Comment by asmeurer** [comment]
@Saanidhyavats when you create a new branch, you need to start with master each time (`git checkout master; git checkout -b branch_name`). Also make sure you never commit to master so that it stays as the upstream master.

**Comment by Saanidhyavats** [comment]
@asmeurer I have updated the suggested changes. I have already created a pull request for changes in files like `function.py` and `vector.py.` In this request I want to only concentrate on file `refine.py`

**Comment by sylee957** [comment]
I don't think that this is properly working and this lacks tests.
I'm not sure any code are copied from #16258, but I think that #16258 is better than this.
I think that it's better for you to collaborate and resolve the conflicts on #16258, if you want to get your name listed on authors.

File: sympy/assumptions/refine.py [hunk 3]
```diff
@@ -336,6 +337,26 @@ def refine_sign(expr, assumptions):
...
 
 
+def refine_matrixelement(expr, assumptions):
+    """
+    Examples
+    ========
+        >>>refine(X[0, 1], Q.symmetric(X))
+        X[0, 1]
+        >>> refine(X[1, 0], Q.symmetric(X))
+        X[0,1]
```
Comment (czgdp1807) [inline sympy/assumptions/refine.py]: Fix the indentation here.

File: sympy/assumptions/refine.py [hunk 4]
```diff
@@ -336,6 +337,26 @@ def refine_sign(expr, assumptions):
...
+    """ 
+    arg = expr.args[0]
+    if ask(Q.symmetric(arg), assumptions):
+        if (expr.args[1] > expr.args[2]):
```
Comment (czgdp1807) [inline sympy/assumptions/refine.py]: What about `if ask(Q.symmetric(arg), assumptions) and (expr.args[1] > expr.args[2]):`?
Is it possible for `expr.args[1]` or `expr.args[2]` to be symbolic? How it works in that case?

File: sympy/assumptions/refine.py [hunk 5]
```diff
@@ -336,6 +337,26 @@ def refine_sign(expr, assumptions):
...
+            temp = expr.args[1]
+            expr.args[1] = expr.args[2]
+            expr.args[2] = temp
+            return(expr)
+        return(expr)
```
Comment (czgdp1807) [inline sympy/assumptions/refine.py]: `return(expr) -> return expr`

**Comment by Saanidhyavats** [comment]
I was thinking to update the changes in this pull request and get it merged( I have taken reference from other functions in same file to create the function `refine_matrixelement`). I have not looked over #16258. Soon I will update the changes  in `refine_matrixelement` and add the test files . I am thinking to delete other changes made in other files which by mistake came in this pr.

File: sympy/assumptions/refine.py (hunk 4 above)
Comment (Saanidhyavats) [reply sympy/assumptions/refine.py]: `refine_matrixelement(X[0, 1], Q.symmetric(X))`.
here expr.args[1] produces 0 and expr.args[2] produces 1.  and expected result should be `x[0,1]`

**Comment by Saanidhyavats** [comment]
I have added the test file and made some additional changes in the file. Correct me if I am wrong somewhere @sylee957 @czgdp1807 @asmeurer .

**Comment by oscarbenjamin** [comment]
@Saanidhyavats can you give this PR a better title?

**Comment by sylee957** [comment]
I see you still have to revert the changes of `function.py`, `vector.py`, `dyadic.py`

**Comment by Saanidhyavats** [comment]
@sylee957  I am trying to revert the changes. Whether deleting them from this request is a correct option?

**Comment by sylee957** [comment]
I’d put them back as the original file. But things would be much easier if you just close this and begin from the scratch, at this point.

**Comment by Saanidhyavats** [comment]
@sylee957 thank you for the suggestion. I have created another pull request(#18681 ) and will be doing further changes in that request. For now I am closing this request.

**Pull request closed without merging**